ingeniamotion/pdo.py

```diff
@@ -0,0 +1,375 @@
+import threading
+import time
+from typing import TYPE_CHECKING, Dict, List, Optional, Union, Tuple, Type, Callable
+
+from ingenialink.pdo import RPDOMap, TPDOMap, RPDOMapItem, TPDOMapItem
+from ingenialink.ethercat.network import EthercatNetwork
+from ingenialink.ethercat.servo import EthercatServo
+from ingenialink.ethercat.register import EthercatRegister
+from ingenialink.canopen.network import CanopenNetwork
+
+from ingeniamotion.exceptions import (
+    IMException,
+)
+from ingeniamotion.metaclass import DEFAULT_AXIS, DEFAULT_SERVO
+from ingeniamotion.enums import COMMUNICATION_TYPE
+
+if TYPE_CHECKING:
+    from ingeniamotion.motion_controller import MotionController
+
+
+class PDONetworkManager:
+    """Manage all the PDO functionalities.
+
+    Attributes:
+        mc: The MotionController.
+
+    """
+
+    class ProcessDataThread(threading.Thread):
+        """Manage the PDO exchange.
+
+        Attributes:
+            net: The EthercatNetwork instance where the PDOs will be active.
+            refresh_rate: Determines how often (seconds) the PDO values will be updated.
```

Comment:
notify_send_process_data and notify_receive_process_data are missing. And they are Args, not Attributes